### PR TITLE
DaoHelper

### PR DESCRIPTION
**Pull request by kirti-d**
add helper functions for parsing objects returned by querying database

### CODE REVIEW

File: src/test/java/com/google/error_analyzer/backend/LogDaoHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    @Before
+    public void setUp() {
+        logDaoHelper = new LogDaoHelper();
+    }
+
+    // hitId
+    @Test
+    public void getIdsOfGivenSearchHits() throws Exception {
+        SearchHit[] searchHits=new SearchHit[]{hit1, hit2};
```
Comment (shudhatmajain) [inline src/test/java/com/google/error_analyzer/backend/LogDaoHelperTest.java:61]: Should we do line 61, 62 in setup?

File: src/test/java/com/google/error_analyzer/backend/LogDaoHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public final class LogDaoHelperTest {
+    private final String field = "name";
+    private LogDaoHelper logDaoHelper;
+    @Mock 
+    SearchHit hit1;
```
Comment (shudhatmajain) [inline src/test/java/com/google/error_analyzer/backend/LogDaoHelperTest.java:45]: Do we need to mock this?
Can't we simply create an object of SearchHit?

File: src/test/java/com/google/error_analyzer/backend/LogDaoHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    private HashMap<String,Object> getSource(String content){
+        HashMap<String,Object> fieldMap = new HashMap();
+        fieldMap.put(field,content);
+        return fieldMap;
+    }
+
+    private HashMap<String, HighlightField> highlight(String highlightText){
+        Text [] text = new Text[]{new Text(highlightText)};
+        HashMap <String,HighlightField> highlight = new HashMap();
+        highlight.put(field,new HighlightField(field,text));
```
Comment (shudhatmajain) [inline src/test/java/com/google/error_analyzer/backend/LogDaoHelperTest.java:116]: highlight.put(field, new HighlightField(field, text));

Spacing.
Similarly at other places.

File: src/test/java/com/google/error_analyzer/backend/LogDaoHelperTest.java (hunk 2 above)
Comment (kirti-d) [reply src/test/java/com/google/error_analyzer/backend/LogDaoHelperTest.java:45]: making SearchHit object increases testing complexities which is not required in our case

**Pull request merged**